### PR TITLE
daemon/main: Remove waitFDPair and related blocking code on it

### PR DESCRIPTION
**Pull request by Xe**
In a Unix-style daemon the parent process should *immediately* exit after it realizes it is set to daemonize. The current behavior was causing the parent process to terminate only after the child was finised loading all of its counters, leading to bizarre behavior when being started using Ubuntu's Upstart.

As is customary with Unix-style daemons, this patch also introduces the child process correctly creating its own process group using the setsid(2) call.

### CODE REVIEW

File: daemon/main.cpp [hunk 1]
```diff
@@ -642,20 +640,30 @@ void handle_pid_daemon(std::string const &pidf, bool daemon)
 
     if (daemon)
     {
-        if (pipe(waitFdPair) < 0)
+        pid = fork();
+        if (pid && pid < 0)
```
Comment (jwatte) [inline daemon/main.cpp]: I dare you to find a case where "pid < 0" is true and "pid" is false :-)

File: daemon/main.cpp [hunk 2]
```diff
@@ -642,20 +640,30 @@ void handle_pid_daemon(std::string const &pidf, bool daemon)
...
+
+        pid_t sid;
+        sid = setsid();
+        if (sid > 0)
+        {
+            LogError << "daemonize: setsid() failed.";
+            exit(1);
+        }
+
+        ::close(0);
```
Comment (jwatte) [inline daemon/main.cpp]: The old code explicitly set these to go to /dev/null, which avoids EBADF when the process actually writes. Don't know if this really matters, but it's a cleanliness thing.

**Comment by jwatte** [comment]
What is this "setsid()" new-fangled-ness? Replacing the controlling terminal by closing 0/1/2 was enough for BSD 4.2 and UNIX 5.2, and that should be enough for anybody!
:-)

File: daemon/main.cpp (hunk 1 above)
Comment (Xe) [inline daemon/main.cpp]: whatever case 

```c++
        if (pid)
        {
            if (pid < 0)
            {
                LogError << "daemonize: fork() failed.";
                exit(1);
            }
        }
```

was written for!

File: daemon/main.cpp (hunk 2 above)
Comment (Xe) [inline daemon/main.cpp]: Most of the time write calls to stdin/out/err are unchecked.

File: daemon/main.cpp (hunk 1 above)
Comment (Xe) [inline daemon/main.cpp]: See #18

**Pull request merged**